Lib/ntpath.py

```diff
@@ -141,17 +154,24 @@ def splitdrive(p):
 
     """
     p = os.fspath(p)
+    if isinstance(p, bytes):
+        sep = b'\\'
+        altsep = b'/'
+        colon = b':'
+    else:
+        sep = '\\'
+        altsep = '/'
+        colon = ':'
+
+    if _path_splitroot:
+        try:
+            d = _path_splitroot(p)[0].rstrip('\\/')
+        except ValueError:
+            pass
```

Comment:
Are you still working on this part? 

I think `_path_splitroot()` should support bytes based on the value of `path->narrow`, like other `os` functions. There's not much to implementing that. It already accepts bytes; it just needs the `PyUnicode_EncodeFSDefault()` call. 

For this case, `rstrip()` the separators to get the drive. Then shift over everything that remains. For example:

```python
    root, tail = _path_splitroot(p)
    drive = root.rstrip(sep + altsep)
    return drive, root[len(drive):] + tail
```

It would be cleaner to rework the fallback code as `ntpath.splitroot()`. Then `ntpath` can switch to using `splitroot()` directly in most cases, without exception. If the OS implementation of `PathCchSkipRoot()` changes significantly enough to make tests fail, the fallback code can be updated to match it.

Reply:
I changed it to shift the slash in `_path_splitroot` from root to tail, because that's pretty easy, but supporting bytes in the return value just leads to lots of code duplication, so left it returning str always.